pkg/build/controller/build/build_controller.go

```diff
@@ -1483,6 +1496,136 @@ func (bc *BuildController) findBuildCAConfigMap(build *buildv1.Build, buildPod *
 	return true, nil
 }
 
+func (bc *BuildController) controllerConfigWorker() {
+	for {
+		if quit := bc.controllerConfigWork(); quit {
+			return
+		}
+	}
+}
+
+func (bc *BuildController) controllerConfigWork() bool {
+	key, quit := bc.controllerConfigQueue.Get()
+	if quit {
+		return true
+	}
+	defer bc.controllerConfigQueue.Done(key)
+
+	err := bc.handleControllerConfig()
+	bc.handleControllerConfigError(err, key)
+	return false
+}
+
+func (bc *BuildController) handleControllerConfig() error {
+	// Make sure that we have synced the controller config store
+	// This will always be false if the controller is not running on an openshift/installer cluster,
+	// and therefore does not have the service cert signing controller.
+	//
+	// TODO: Remove this once all tests run on openshift/installer clusters
+	if !bc.controllerConfigStoreSynced() {
+		glog.V(2).Info("builds.config.openshift.io not synced")
+		glog.V(4).Info("disabling service signing CA injection for builds")
+		bc.injectServiceCA = false
+		return nil
+	}
+	bc.injectServiceCA = true
+	config, err := bc.configLister.Get("cluster")
+	if err != nil && !errors.IsNotFound(err) {
+		return err
+	}
+	if config == nil {
+		bc.additionalTrustedCAData = nil
+		return nil
+	}
+	if len(config.Spec.AdditionalTrustedCA.Name) == 0 {
+		bc.additionalTrustedCAData = nil
+		return nil
+	}
+	additionalCA, err := bc.configMapClient.ConfigMaps("openshift-config").Get(config.Spec.AdditionalTrustedCA.Name, metav1.GetOptions{})
+	if err != nil && !errors.IsNotFound(err) {
+		return err
+	}
+	if additionalCA == nil {
+		bc.additionalTrustedCAData = nil
+		return nil
+	}
+	bc.additionalTrustedCAData = additionalCA.Data
+	return nil
+}
+
+func (bc *BuildController) handleControllerConfigError(err error, key interface{}) {
+	if err != nil {
+		glog.V(2).Infof("Failed to update controller config: %v", err)
+	}
+	bc.controllerConfigQueue.Forget(key)
+}
+
+func (bc *BuildController) controllerConfigAdded(obj interface{}) {
+	bc.controllerConfigChanged()
+}
+
+func (bc *BuildController) controllerConfigUpdated(old, cur interface{}) {
+	bc.controllerConfigChanged()
+}
+
+func (bc *BuildController) controllerConfigDeleted(obj interface{}) {
+	bc.controllerConfigChanged()
+}
+
+func (bc *BuildController) controllerConfigChanged() {
+	bc.controllerConfigQueue.Add("openshift-config")
+}
+
+func (bc *BuildController) configMapAdded(obj interface{}) {
+	configMap, ok := obj.(*corev1.ConfigMap)
+	if !ok {
+		utilruntime.HandleError(fmt.Errorf("object was not a configMap: %+v", obj))
+		return
+	}
+	if configMap == nil {
+		return
+	}
+	if configMap.Namespace != "openshift-config" {
```

Comment:
we should only be watching configmaps in this namespace.. is there a reason for us to watch any others?

Reply:
This lister came from the existing "all kube" informer, see above regarding adding a new informer.